### ISSUE TITLE
[BUG] Notifications dashboard doesn't install correctly in OSD

### ISSUE THREAD

**Issue by derek-ho** [opensearch-project/dashboards-notifications#185]
- build notifications dashboard
- install it into OSD
- See the following error:
```
curl: (7) Failed to connect to localhost port 5601 after 0 ms: Connection refused
{"type":"log","@timestamp":"2024-04-19T18:52:41Z","tags":["fatal","root"],"pid":2551,"message":"Error: Cannot find module '../../public/utils/constants'\nRequire stack:\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/configRoutes.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/index.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/plugin.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/index.js\n- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/src/core/server/
 FATAL  Error: Cannot find module '../../public/utils/constants'
Require stack:
- /home/runner/work/opensearch-dashboards-functional-test/opensearch-dashboards-functional-test/opensearch-dashboards-3.0.0/plugins/notificationsDashboards/server/routes/configRoutes.js
- 
```

Related PR: 
https://github.com/opensearch-project/dashboards-notifications/pull/165

Discovered via: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1146

**Comment by ruanyl** [comment]
I'm getting the same error, this seems related a recent commit https://github.com/opensearch-project/dashboards-notifications/commit/1fc2abe548017647b2e624ab9f4854e96e3d724d

**Comment by ruanyl** [comment]
cc @SuZhou-Joe @Hailong-am

**Comment by SuZhou-Joe** [comment]
Let me see.

**Comment by SuZhou-Joe** [comment]
@amsiglan Code from server directory should not rely on the code from public directory.

<img width="500" alt="image" src="https://github.com/opensearch-project/dashboards-notifications/assets/13493605/0e0d2acf-d1c9-48dd-90df-663e57205a92">

For the final build tar, the public directory of plugins will be bundled to `target/{pluginId}.bundle.js`, and server side's code won't be able to find the reference within public directory.

**Comment by Hailong-am** [comment]
this fixed by #189
